pkg/builder/network.go

```diff
@@ -69,6 +69,20 @@ func (v *VMBuilder) Interface(interfaceName, interfaceModel, interfaceMACAddress
 	return v
 }
 
+func (v *VMBuilder) SetNetworkInterfaceBootOrder(interfaceName string, bootOrder uint) *VMBuilder {
+	interfaces := v.VirtualMachine.Spec.Template.Spec.Domain.Devices.Interfaces
+
+	for i, iface := range interfaces {
+		if iface.Name == interfaceName {
+			ifaceCopy := iface.DeepCopy()
+			ifaceCopy.BootOrder = func() *uint { v := uint(bootOrder); return &v }()
```

Comment:
This construction looks difficult. Would it make sense to simplify this, as @w13915984028 does in his code.

```suggestion
			bo := bootOrder
			ifaceCopy.BootOrder = &bo
```

Reply:
If you say so...

Golang (like most programming languages) allocates local variables in a function on the stack by default. But it does recognize when a variable should live beyond the scope of the function, in which case it allocates the variable on the heap instead.

This is a pretty standard way in to force the allocation on the heap and get a pointer to it, which is necessary here because `iface.BootOrder` is a pointer and needs to point to a valid location in the heap.

Since the lambda will be inlined, there is no performance penalty.

The method Jian proposed in his code snipped also works, but it splits the allocation and assignment over two lines, which I think makes the code less clear.
There is also a way to (ab)use arrays to do the same, which is more compact, but I find the type signature on that one harder to read.